osquery/filesystem/filesystem.cpp

```diff
@@ -147,6 +147,136 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
   }
 }
 
+Status listDirectoriesInDirectory(const boost::filesystem::path& path,
+                            std::vector<std::string>& results) {
+  try {
+    if (!boost::filesystem::exists(path)) {
+      return Status(1, "Directory not found");
+    }
+
+    if (!boost::filesystem::is_directory(path)) {
+      return Status(1, "Supplied path is not a directory");
+    }
+
+    boost::filesystem::directory_iterator begin_iter(path);
+    boost::filesystem::directory_iterator end_iter;
+    for (; begin_iter != end_iter; begin_iter++) {
+      if (boost::filesystem::is_directory(begin_iter->path())) {
+        results.push_back(begin_iter->path().string());
+      }
+    }
+
+    return Status(0, "OK");
+  } catch (const boost::filesystem::filesystem_error& e) {
+    return Status(1, e.what());
+  }
+}
+
+std::vector<std::string> &split(const std::string &s, char delim, std::vector<std::string> &elems) {
+    std::stringstream ss(s);
+    std::string item;
+    while (std::getline(ss, item, delim)) {
+      if(item.length() == 0){continue;}
+      elems.push_back(item);
+    }
+    return elems;
+}
+
+std::string crush(int start, int end, std::vector<std::string> components){
```

Comment:
Use `boost::algorithm::join`: http://stackoverflow.com/questions/1833447/a-good-example-for-boostalgorithmjoin